IBM/carbon-addons-iot-react
# Update Storybook to v5.3.13

Closes #554 

**Summary**

- Upgrade Storybook version ([read more here](https://medium.com/storybookjs/storybook-5-3-83e114e8797c)), update the config based on the [migration guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#migration), migrate one story from `storiesOf` to CSF.
- A few of the commits here have extra context in the commit body for reference
- Unfortunately it was unavoidable to touch every story and therefore every storyshot

**Change List (commits, features, bugs, etc)**

- upgrade relevant deps to 5.3.13
- rename/remove/refactor configuration files to new names and formats (`main.js`, `preview.js`).
- refactor webpack config to use `full-control` mode instead of deprecated `extend` mode
- refactor stories to use the recommended root/hierarchy separator, `\` instead of previous `|`
- refactor all imported/reexported carbon stories to use a named export - [it's a workaround](https://github.com/storybookjs/storybook/issues/9196#issuecomment-567737450) because they're using the old `storiesOf` API
- modify jest config to avoid an esm import issue and transform CSF story names
- increase heap size for static storybook (prod) builds - it times out locally due to an issue that should be fixed in Storybook v6.
- remove TitleAddon - is no longer needed
- refactor Welcome story to use CSF format instead of old `storiesOf` api
- update snapshots

**Acceptance Test (how to verify the PR)**

- View all the stories in the storybook, verify they render as intended.
- Ensure knobs and other storybook addons/functions are working as expected.
